src/emitter/emitter.ts

```diff
@@ -1,28 +1,28 @@
-import StrictEventEmitter from 'strict-event-emitter-types';
 import Emitter = require('wolfy87-eventemitter');
 import { createDisposable, IDisposable } from '../disposable';
 
-export interface IEmitGroup<E extends object> {
+export type EventMap = {
+  [key: string]: (...args: any) => void;
```

Comment:
Note that as a quirk of TypeScript, index types of `[key: string]` also support number. (source: go look it up. But it's because in JS, numeric indices are implicitly converted to string keys anyway according to the spec #shudder.) I've yalced this into Platform and can confirm the typings all work nicely with the numeric `enum`s that are occasionally used (I'm looking at you, graphite).

Reply:
> But it's because in JS, numeric indices are implicitly converted to string keys anyway according to the spec 

Yeah, I never actually understood why TS supports `number` as PropertyKey. A number cannot be a valid property key in JS, so it doesn't make sense.
They should only support strings & symbols.

> (I'm looking at you, graphite).

:see_no_evil:

Reply:
> > (I'm looking at you, graphite).
> 
> 🙈

ooh look, JsonPatch event `11`... (I go and counting all the lines in the Enum to figure out whatevertheheck that means 😆 )